### PR TITLE
Move DownloadConnect to shared

### PR DESCRIPTION
**Pull request by avatus**
This PR moves the `DownloadConnect` component to the `shared` package. It is going to be reused in the device trust webUI as well so I figured I could take it out of Discover

### CODE REVIEW

File: web/packages/shared/components/DownloadConnect/DownloadConnect.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ */
+
+import { ButtonSecondary } from 'design/Button';
+
+import { MenuButton, MenuItem } from 'shared/components/MenuAction';
+
+export type DownloadLink = { text: string; url: string };
+
+export const DownloadConnect = (props: {
+  downloadLinks: Array<DownloadLink>;
```
Comment (gzdunek) [inline web/packages/shared/components/DownloadConnect/DownloadConnect.tsx:26]: Nit:
```suggestion
  downloadLinks: DownloadLink[];
```

File: web/packages/shared/components/DownloadConnect/DownloadConnect.tsx
Comment (gzdunek) [inline web/packages/shared/components/DownloadConnect/DownloadConnect.tsx:1]: Maybe we could put it in `web/packages/shared/components/DownloadConnect.tsx`? 

I think there is no need to keep a single file in its own directory.

**Review by gzdunek — APPROVED** [review]

File: web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.tsx [hunk 2]
```diff
@@ -320,29 +323,6 @@ export const usePollForConnectMyComputerNode = (args: {
...
-      {props.downloadLinks.map(link => (
-        <MenuItem key={link.url} as="a" href={link.url}>
-          {link.text}
-        </MenuItem>
-      ))}
-    </MenuButton>
-  );
-};
-
 function getConnectDownloadLinks(
```
Comment (ravicious) [inline web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.tsx:326]: Would it make sense to extract `getConnectDownloadLinks` into `DownloadConnect.tsx` as well? I imagine the logic to get a link will be pretty much the same, no?

File: web/packages/shared/components/DownloadConnect/DownloadConnect.tsx
Comment (ravicious) [inline web/packages/shared/components/DownloadConnect/DownloadConnect.tsx:1]: Could you add a small story file which basically shows `DownloadConnect` with a single link and multiple links? Technically this is already covered by the story for `SetupConnect`, but the logic there is a bit different (how the component looks depends on the user agent). Here the story would be much simpler.

I know it's additional work, but I feel like when we're moving components to shared, it's ultimately worth it, as those components are meant to be used in multiple places.

**Review by ravicious — APPROVED** [review]

File: web/packages/shared/components/DownloadConnect/DownloadConnect.tsx
Comment (gzdunek) [reply web/packages/shared/components/DownloadConnect/DownloadConnect.tsx:1]: (note: if we add a story file, then my comment above is no longer relevant)

Comment (avatus) [reply web/packages/shared/components/DownloadConnect/DownloadConnect.tsx:1]: Added storyfile with links for each platform (which takes care of single/multiple links)

File: web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.tsx (hunk 2 above)
Comment (avatus) [reply web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.tsx]: yes, good call. moved into the file as well

**Pull request merged**